### ISSUE TITLE
gatsby-remark-prismjs adding newlines to inline code blocks

### ISSUE THREAD

**Issue by adamyonk** [gatsbyjs/gatsby#4802]
## Description

I'm seeing a newline added at the end of inline code blocks, but *only* if there is a `inlineCodeMarker` present.

![image](https://user-images.githubusercontent.com/33258/38204814-f0438fbc-3669-11e8-881d-35de609e9996.png)
![image](https://user-images.githubusercontent.com/33258/38204831-fb639090-3669-11e8-88e6-a0f0ce28cdb7.png)

### Steps to reproduce

Here's the markdown in question:
```
When Jekyll picks up my built bundle to compile it into `/_site`, it will now
have the triple-dashes prepended to the file. I use the `style-loader` with
Webpack so that I can `javascript±require` styles in my JavaScript and have
them written to the document `<head>`. So now I'll be able to use Liquid tags
for things like this:
```

### Expected result

No newline added after inline code with `inlineCodeMarker`.

### Actual result

Newline added after inline code.

### Environment

* Gatsby version (`npm list gatsby`): `1.9.243`
* gatsby-cli version (`gatsby --version`): `1.9.243`
* Node.js version: `8.9.4`
* Operating System: `macOS 10.13.3`

### File contents (if changed):

`gatsby-config.js`:
```javascript
module.exports = {
  siteMetadata: {
    title: "Adam Jahnke ☕️🏍 (adamyonk)",
    description: "",
    siteUrl: "https://adamyonk.com",
    author: "Adam Jahnke",
  },
  plugins: [
    { resolve: "gatsby-plugin-react-next" },
    { resolve: "gatsby-plugin-react-helmet" },
    { resolve: "gatsby-plugin-styled-jsx" },
    {
      resolve: "gatsby-plugin-feed-generator",
      feedQuery: `
        {
          allMarkdownRemark(
            filter: { frontmatter: { templateKey: { eq: "post" } } }
            sort: { order: DESC, fields: [frontmatter___date] }
          ) {
            edges {
              node {
                html
                id
                frontmatter {
                  date
                  path
                  title
                }
              }
            }
          }
        }
      `,
    },
    {
      resolve: "gatsby-source-filesystem",
      options: {
        path: `${__dirname}/src/pages`,
        name: "pages",
      },
    },
    {
      resolve: "gatsby-transformer-remark",
      options: {
        plugins: [
          {
            resolve: "gatsby-remark-prismjs",
            options: {
              inlineCodeMarker: "±",
            },
          },
          {
            resolve: "gatsby-remark-images",
            options: {
              maxWidth: 800,
              linkImagesToOriginal: true,
            },
          },
        ],
      },
    },
  ],
}
```

`package.json`:
```json
{
  "name": "adamyonk.com",
  "description": "",
  "version": "1.0.0",
  "author": "Adam Jahnke",
  "dependencies": {
    "date-fns": "^1.29.0",
    "gatsby": "^1.9.243",
    "gatsby-link": "^1.6.40",
    "gatsby-plugin-feed-generator": "^1.0.5",
    "gatsby-plugin-react-helmet": "^1.0.8",
    "gatsby-plugin-react-next": "^1.0.11",
    "gatsby-plugin-styled-jsx": "^2.0.6",
    "gatsby-remark-images": "^1.5.60",
    "gatsby-remark-prismjs": "^1.2.21",
    "gatsby-source-filesystem": "^1.5.27",
    "gatsby-transformer-remark": "^1.7.39",
    "prettier": "^1.11.1",
    "styled-jsx": "^2.2.6"
  },
  "private": true,
  "main": "n/a",
  "scripts": {
    "build": "gatsby build",
    "start": "gatsby develop",
    "format": "prettier './*.{js,json}' '.{babel,prettier}rc' 'src/**/*.{js,json,md}' --write ",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "devDependencies": {}
}
```

`gatsby-node.js`:
```javascript
const path = require("path")

exports.createPages = ({ boundActionCreators, graphql }) => {
  const { createPage } = boundActionCreators

  return graphql(`
    {
      pages: allMarkdownRemark(
        sort: { order: DESC, fields: [frontmatter___date] }
        limit: 1000
      ) {
        edges {
          node {
            excerpt(pruneLength: 400)
            html
            id
            frontmatter {
              templateKey
              path
              date
              title
            }
          }
        }
      }
      posts: allMarkdownRemark(
        filter: { frontmatter: { templateKey: { eq: "post" } } }
        limit: 1000
      ) {
        edges {
          node {
            frontmatter {
              tags
            }
          }
        }
      }
    }
  `).then(result => {
    if (result.errors) {
      return Promise.reject(result.errors)
    }
    result.data.posts.edges
      .reduce((a, { node: { frontmatter: { tags } } }) => {
        for (const tag of tags) {
          a.add(tag)
        }
        return a
      }, new Set())
      .forEach(tag => {
        createPage({
          path: `/tags/${tag}`,
          component: path.resolve(`src/templates/tag.js`),
          context: { tag }, // additional data can be passed via context
        })
      })
    result.data.pages.edges.forEach(({ node }) => {
      createPage({
        path: node.frontmatter.path,
        component: path.resolve(
          `src/templates/${String(node.frontmatter.templateKey)}.js`
        ),
        context: {}, // additional data can be passed via context
      })
    })
  })
}
```

`gatsby-browser.js`: 
`gatsby-ssr.js`:

**Comment by KyleAMathews** [comment]
@docwhat any idea?

@adamyonk this probably happened in https://github.com/gatsbyjs/gatsby/pull/4556

**Comment by adamyonk** [comment]
I should also note, I'm setting this up for the first time, I'm not sure if past versions of this worked or not.

**Comment by docwhat** [comment]
With the default CSS setup, it shouldn't matter if there is a newline in an inline `<code>` element.

As for why a trailing newline is added, that _is_ weird.  My [code](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-prismjs/src/index.js#L64-L67) doesn't add a new line.  The Remark parser shouldn't be adding a newline to `node.value`.

**Comment by docwhat** [comment]
I'm not sure if this is a bug or not; PrismJS CSS has rules for non-pre-wrapped `code` that change the whitespace rules so this doesn't matter.

You can see the extra newline being added [on my syntax test page](https://docwhat.org/pi/markup-syntax-highlighting#inlinecode-highligthing):

Example:

```markdown
`javascript±const reallyLongNameThatIsAnnoyinglyLong = { apples: 1, cats: 3, dogs: 20, deer: 2, rodents: 5 };`
```

Is turned into:

```html
<code class="language-javascript"><span class="token keyword">const</span> reallyLongNameThatIsAnnoyinglyLong <span class="token operator">=</span> <span class="token punctuation">{</span> apples<span class="token punctuation">:</span> <span class="token number">1</span><span class="token punctuation">,</span> cats<span class="token punctuation">:</span> <span class="token number">3</span><span class="token punctuation">,</span> dogs<span class="token punctuation">:</span> <span class="token number">20</span><span class="token punctuation">,</span> deer<span class="token punctuation">:</span> <span class="token number">2</span><span class="token punctuation">,</span> rodents<span class="token punctuation">:</span> <span class="token number">5</span> <span class="token punctuation">}</span><span class="token punctuation">;</span>
</code>
```

The the trailing newline after the last `</span>`.

**Comment by docwhat** [comment]
Hey, @KyleAMathews:

Could this be the source of the extra newline?

<https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-prismjs/src/highlight-code.js#L42>

**Comment by adamyonk** [comment]
Ok, thanks for the information. I'll try to isolate the styles for this some more to make sure I don't have something conflicting.

**Comment by docwhat** [comment]
I think [this](https://github.com/gatsbyjs/gatsby/blob/6c120dbc43e4e1ade0e43f5ac5f8418071a7e00d/packages/gatsby-remark-prismjs/src/highlight-code.js#L36-L44) should be changed to:

```javascript
codeSplits.forEach(split => {
  highlightedCode += `${split.code}`
})
```

**Comment by adamyonk** [comment]
> ```javascript
> codeSplits.forEach(split => {
>   highlightedCode += `${split.code}`
> })
> ```

That seems like it would fix my problem. Can confirm that none of my own styles are causing this.

**Comment by docwhat** [comment]
It actually doesn't.  It'll collapse non-highlighted newline.

I did a fix in #4804 and if it passes all the tests, I think it is correct.

**Comment by adamyonk** [comment]
🎊
